src/Umbraco.Web.Common/Controllers/UmbracoPageController.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ViewEngines;
+using Microsoft.Extensions.Logging;
+using Umbraco.Core.Models.PublishedContent;
+using Umbraco.Web.Common.Routing;
+using Umbraco.Web.Routing;
+
+namespace Umbraco.Web.Common.Controllers
+{
+    /// <summary>
+    /// An abstract controller for a front-end Umbraco page
+    /// </summary>
+    public abstract class UmbracoPageController : UmbracoController
+    {
+        private UmbracoRouteValues _umbracoRouteValues;
+        private readonly ICompositeViewEngine _compositeViewEngine;
+        private readonly ILogger<UmbracoPageController> _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UmbracoPageController"/> class.
+        /// </summary>
+        protected UmbracoPageController(ILoggerFactory loggerFactory, ICompositeViewEngine compositeViewEngine)
```

Comment:
Not sure why we should use an `ILoggerFactory` and not just `ILogger<UmbracoPageController>`?

Remember the definition is `ILogger<out T>`, so `ILogger<RenderController>` can be injected into the `ILogger<UmbracoPageController>` as long as `RenderController` is a `UmbracoPageController`